utilities/pytest_utils.py

```
@@ -52,6 +55,72 @@
+    name = sanitized_name or "execution_failure"
+
+    node_id = f"pytest_exit::{name}"
+    reporter = xml_writer.node_reporter(report=node_id)
+    reporter.attrs["classname"] = "pytest_exit"
+    reporter.attrs["name"] = name
+
+    error_node = ElementTree.Element("error", attrib={"message": f"Pytest execution failed (exit code: {return_code})"})
+    error_node.text = bin_xml_escape(arg=log_message)
+    reporter.append(error_node)
+
+    LOGGER.info(f"Injected synthetic failure testcase into JUnit XML (exit code: {return_code})")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)